### ISSUE TITLE
Windows - Graph Apps Gallery - Issue when installing

### ISSUE THREAD

**Issue by MichaelRBlack** [neo4j-apps/neo4j-desktop#60]
Unable to install the Graph Apps Gallery

[](https://install.graphapp.io/)

```
[2019-04-08 12:32:51:0853] [info] Function[]
[2019-04-08 12:32:51:0853] [info] Installing graph app from: https://install.graphapp.io/
[2019-04-08 12:32:51:0860] [info] Action[GRAPH_APP_INSTALLING] MemoryInfo {
  graphApp:
   MemoryInfo {
     graphAppUrl: 'https://install.graphapp.io/',
     id: '73c05381-e92d-471d-91b0-7ee8be245b60' },
  type: 'GRAPH_APP_INSTALLING' }
[2019-04-08 12:32:52:0748] [error] Graph app installation failed Error: C:\Neo4jDatabases\graphApps\Neo4j Desktop Apps Gallery: One-Click Install of Curated Graph Developer Tools contains invalid WIN32 path characters.
    at mkdirs (C:\Users\mblack\AppData\Local\Programs\neo4j-desktop\resources\app.asar\node_modules\fs-extra\lib\mkdirs\mkdirs.js:18:22)
    at Promise (C:\Users\mblack\AppData\Local\Programs\neo4j-desktop\resources\app.asar\node_modules\universalify\index.js:13:12)
    at new Promise (<anonymous>)
    at mkdirs (C:\Users\mblack\AppData\Local\Programs\neo4j-desktop\resources\app.asar\node_modules\universalify\index.js:7:14)
    at t.default.addGraphApp (C:\Users\mblack\AppData\Local\Programs\neo4j-desktop\resources\app.asar\dist\main.prod.js:1:92495)
    at process._tickCallback (internal/process/next_tick.js:68:7)

[2019-04-08 12:32:52:0751] [info] Action[GRAPH_APP_INSTALLING] MemoryInfo {
  graphApp:
   MemoryInfo {
     appId:
      'Neo4j Desktop Apps Gallery: One-Click Install of Curated Graph Developer Tools',
     appName:
      'Neo4j Desktop Apps Gallery: One Click Install Of Curated Graph Developer Tools',
     graphAppType: 'REMOTE',
     graphAppUrl: 'https://install.graphapp.io/',
     id: '73c05381-e92d-471d-91b0-7ee8be245b60' },
  type: 'GRAPH_APP_INSTALLING' }
[2019-04-08 12:32:52:0791] [info] Action[GRAPH_APP_INSTALL_FAILED] MemoryInfo {
  error:
   MemoryInfo {
     code: 'EINVAL',
     message:
      'C:\\Neo4jDatabases\\graphApps\\Neo4j Desktop Apps Gallery: One-Click Install of Curated Graph Developer Tools contains invalid WIN32 path characters.',
     name: 'Error' },
  graphApp:
   MemoryInfo {
     graphAppUrl: 'https://install.graphapp.io/',
     id: '73c05381-e92d-471d-91b0-7ee8be245b60' },
  type: 'GRAPH_APP_INSTALL_FAILED' }
```

**Comment by jexp** [comment]
I changed the title of the page, can you try again?

For the desktop team, we should probably not use the page-title 1:1 as file name as it doesn't have the same restrictions as the npm app-id.

**Comment by oskarhane** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Generate safe appIds for hosted graph apps](https://trello.com/c/PQy30WU6/920-generate-safe-appids-for-hosted-graph-apps)

**Comment by oskarhane** [comment]
Thanks for the report, this looks like a sanitization bug. We'll have a look at it.

**Comment by MichaelRBlack** [comment]
@jexp I tried again and instead of an error, the spinner just sat there attempting to install and load for a long time until I gave up and killed the desktop application.

**Comment by jexp** [comment]
Which desktop version are you using?
I tried to reproduce on OSX and it worked.

**Comment by MichaelRBlack** [comment]
https://install.graphapp.io/

![image](https://user-images.githubusercontent.com/4128408/55894029-9b394100-5b76-11e9-9d23-e748ca1f3063.png)

I think what may be happening is a certificate error. I just noticed when I tried browsing to the site in a web browser it gave me a warning about the cert. Probably explains why desktop just hung trying to load. Maybe a feature improve suggestion for the desktop app to exit when there's a cert error.

![image](https://user-images.githubusercontent.com/4128408/55894259-0aaf3080-5b77-11e9-83f7-e68440a3e1dc.png)
